docs/how-to-guides/data-basics/3-building-an-edit-form.md

```diff
@@ -0,0 +1,521 @@
+
+```js
+export function VanillaReactForm({ initialTitle }) {
+	const [title, setTitle] = useState( initialTitle );
+	return (
+		<TextControl
+			value={ title }
+			onChange={ setTitle }
+		/>
+	);
+}
+```
+
+Updating entity records in Gutenberg Data is similar but instead of using `setTitle` to store in local state, we use the `editEntityRecord` action which stores the updates in the _Redux_ state. Here's how you can try it out in your browser's dev tools:
+
+```js
+// We need a valid page ID to call editEntityRecord, so let's get the first available one using getEntityRecords.
+const pageId = wp.data.select( 'core' ).getEntityRecords( 'postType', 'page' )[0].id;
+
+// Update the title
+wp.data.dispatch( 'core' ).editEntityRecord( 'postType', 'page', pageId, { title: 'updated title' } );
+```
+
+At this point, you may ask _how is `editEntityRecord` better than `useState`? The answer is that it offers a few features you wouldn't otherwise get.
+
+Firstly, we can save the changes as easily as we retrieve the data and ensure that all caches will be correctly updated.
+
+Secondly, the changes applied via `editEntityRecord` are easily undo-able via the `undo` and `redo` actions.
+
+Lastly, because the changes live in the _Redux_ state, they are "global" and can be accessed by other components. For example, we could make the `PagesList` display the currently edited title.
+
+To that last point, let's see what happens when we use `getEntityRecord` to access the entity record we just updated:
+
+```js
+wp.data.select( 'core' ).getEntityRecord( 'postType', 'page', pageId ).title
+```
+
+It doesn't reflect the edits. What's going on?
+
+Well, let's imagine for a second that it would reflect the edits. If it did, then anything you typed in the title field would be _immediately_ displayed inside the `PagesList`.
+
+Gutenberg Data distinguishes between *Entity Records* and *Edited Entity Records*. *Entity Records* reflect the data from the API and ignore any local edits, while *Edited Entity Records* also have all the local edits applied on top. Both co-exist in the Redux state at the same time.
+
+Let's see what happens if we call `getEditedEntityRecord`:
+
+```js
+wp.data.select( 'core' ).getEditedEntityRecord( 'postType', 'page', pageId ).title
+// "updated title"
+
+wp.data.select( 'core' ).getEntityRecord( 'postType', 'page', pageId ).title
+// { "rendered": "<original, unchanged title>", "raw": "..." }
+```
+
+As you can see, the `title` of an Entity Record is an object, but the `title` of an Edited Entity record is a string.
+
+This is no accident. Fields like `title`, `excerpt`, and `content` can only be rendered on the server so the REST API exposes both the `raw` markup and the `rendered` string. For example, in the block editor, `content.rendered` could used as a visual preview, and `content.raw` could be used to populate the code editor.
```

Comment:
> can only be rendered on the server

How so? I wasn't clear here.

Reply:
Thanks for noting! I'll revise this using a few lines from the [TypeScript types PR](https://github.com/WordPress/gutenberg/pull/38666/) where this is explained a bit more clearly.